libs/dyn/convert/from_typed.go

```diff
@@ -126,7 +124,7 @@ func fromTypedStruct(src reflect.Value, ref dyn.Value, options ...fromTypedOptio
 
 		// Either if the key was set in the reference or the field is not zero-valued, we include it.
 		if ok || nv.Kind() != dyn.KindNil {
-			out.Set(refk, nv) // nolint:errcheck
+			out.SetLoc(k, nil, nv)
```

Comment:
This conversion erases key locations if the key already exists in the reference.

It needs `refloc` to store pre-existing locations from `pair.Key`.

If no test is failing, we're missing coverage for this.

Reply:
Thanks, added 9867a66a01757686b5963d9ba65911cc5acf3065